core/tauri/src/menu/plugin.rs

```diff
@@ -0,0 +1,637 @@
+  app: AppHandle<R>,
+  rid: ResourceId,
+  kind: ItemKind,
+) -> crate::Result<bool> {
+  let resources_table = app.manager.resources_table();
+  do_menu_item!(|i| i.is_enabled(), !Predefined)
+}
+
+#[command(root = "crate")]
+fn set_enabled<R: Runtime>(
+  app: AppHandle<R>,
+  rid: ResourceId,
+  kind: ItemKind,
+  enabled: bool,
+) -> crate::Result<()> {
+  let resources_table = app.manager.resources_table();
+  do_menu_item!(|i| i.set_enabled(enabled), !Predefined)
+}
+
+#[command(root = "crate")]
+fn set_accelerator<R: Runtime>(
+  app: AppHandle<R>,
+  rid: ResourceId,
+  kind: ItemKind,
+  accelerator: Option<String>,
+) -> crate::Result<()> {
+  let resources_table = app.manager.resources_table();
+  do_menu_item!(|i| i.set_accelerator(accelerator), !Predefined | !Submenu)
+}
+
+#[command(root = "crate")]
+fn set_as_windows_menu_for_nsapp<R: Runtime>(
+  app: AppHandle<R>,
+  rid: ResourceId,
+) -> crate::Result<()> {
+  #[cfg(target_os = "macos")]
+  {
+    let resources_table = app.manager.resources_table();
+    let submenu = resources_table.get::<Submenu<R>>(rid)?;
+    submenu.set_as_help_menu_for_nsapp()?;
+  }
+
+  let _ = rid;
+  let _ = app;
+  Ok(())
+}
+
+#[command(root = "crate")]
+fn set_as_help_menu_for_nsapp<R: Runtime>(app: AppHandle<R>, rid: ResourceId) -> crate::Result<()> {
+  #[cfg(target_os = "macos")]
+  {
+    let resources_table = app.manager.resources_table();
+    let submenu = resources_table.get::<Submenu<R>>(rid)?;
+    submenu.set_as_help_menu_for_nsapp()?;
+  }
+
+  let _ = rid;
+  let _ = app;
+
+  Ok(())
+}
+
+#[command(root = "crate")]
+fn is_checked<R: Runtime>(app: AppHandle<R>, rid: ResourceId) -> crate::Result<bool> {
+  let resources_table = app.manager.resources_table();
+  let check_item = resources_table.get::<CheckMenuItem<R>>(rid)?;
+  check_item.is_checked()
+}
+
+#[command(root = "crate")]
+fn set_checked<R: Runtime>(app: AppHandle<R>, rid: ResourceId, checked: bool) -> crate::Result<()> {
+  let resources_table = app.manager.resources_table();
+  let check_item = resources_table.get::<CheckMenuItem<R>>(rid)?;
+  check_item.set_checked(checked)
+}
+
+#[command(root = "crate")]
+fn set_icon<R: Runtime>(
+  app: AppHandle<R>,
+  rid: ResourceId,
+  icon: Option<IconDto>,
+) -> crate::Result<()> {
+  let resources_table = app.manager.resources_table();
+  let icon_item = resources_table.get::<IconMenuItem<R>>(rid)?;
+  icon_item.set_icon(icon.map(Into::into))
+}
+
+#[command(root = "crate")]
+fn set_native_icon<R: Runtime>(
+  app: AppHandle<R>,
+  rid: ResourceId,
+  icon: Option<NativeIcon>,
+) -> crate::Result<()> {
+  let resources_table = app.manager.resources_table();
+  let icon_item = resources_table.get::<IconMenuItem<R>>(rid)?;
+  icon_item.set_native_icon(icon)
+}
+
+pub(crate) fn init<R: Runtime>() -> TauriPlugin<R> {
+  Builder::new("menu")
+    .on_event(|app, e| {
+      if let RunEvent::MenuEvent(e) = e {
+        let _ = app.emit_all("tauri://menu", e);
```

Comment:
consider using a channel instead (maybe have a dedicated list of channels in the resources table)